feat: Mint nft with collection address

xwxtwd opened the pull request:
<!-- Use this template by filling in information and copy and pasting relevant items out of the html comments. -->

# Relates to:
N/A
<!-- LINK TO ISSUE OR TICKET -->

<!-- This risks section is to be filled out before final review and merge. -->

# Risks

<!--
Low, medium, large. List what kind of risks, and what could be effected.
-->
Low
# Background

## What does this PR do?
Add a new action to plugin-nft-generation that allows minting NFTs to a specified collection address.

## What kind of change is this?
Features (non-breaking change which adds functionality)

<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->

<!-- This "Why" section is most relevant if there is no linked issue explaining why. If there is a related issue it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->

# Documentation changes needed?
My changes do not require a change to the project documentation.

<!--
My changes do not require a change to the project documentation.
My changes require a change to the project documentation.
If a docs change is needed: I have updated the documentation accordingly.
-->

<!-- Please show how you tested the PR. This will really help if the PR needs to be retested, and probably help the PR get merged quicker. -->

# Testing

## Where should a reviewer start?
- Check the new action implementation in `plugin-nft-generation/src/actions/mintNFTAction.ts`
- Review the action registration in `plugin-nft-generation/src/index.ts`

## Detailed testing steps
1. Ask the bot: create collection
2. Ask the bot: Help me mint an NFT 
<!--
None, automated tests are fine.
-->

<!--
- As [anon/admin], go to [link]
  - [do action]
  - verify [result]
-->

<!-- If there is a UI change, please include before and after screenshots or videos. This will speed up PRs being merged. It is extra nice to annotate screenshots with arrows or boxes pointing out the differences. -->
<!--
## Screenshots
### Before
### After
-->

<!-- If there is anything about the deploy, please make a note. -->
<!--
# Deploy Notes
-->

<!--  Copy and paste commandline output. -->
<!--
## Database changes
-->

<!--  If there is something more than the automated steps, please specifiy deploy instructions. -->
<!--
## Deployment instructions
-->

<!-- If you are on Discord, please join https://discord.gg/ai16z and state your Discord username here for contribute role and join us in #development-feed -->
<!--
## Discord username

-->

monilpat commented on packages/plugin-nft-generation/src/actions/mintNFTAction.ts:
```diff
@@ -0,0 +1,338 @@
+import WalletSolana from "../provider/wallet/walletSolana.ts";
+import { PublicKey } from "@solana/web3.js";
+
+const mintTemplate = `Respond with a JSON markdown block containing only the extracted values. Use null for any values that cannot be determined.
```
Thanks for doing this - would consider creating a template file and moving this over as is the norm for other plugins

xwxtwd replied:
Alright, I'll get right on it.

monilpat commented on packages/plugin-nft-generation/src/actions/mintNFTAction.ts:
```diff
@@ -0,0 +1,338 @@
+    ],
+    description: "Mint NFTs for the collection",
+    validate: async (runtime: IAgentRuntime, _message: Memory) => {
+        const AwsAccessKeyIdOk = !!runtime.getSetting("AWS_ACCESS_KEY_ID");
```
naming convention camelCase here

xwxtwd replied:
fixed

monilpat commented on packages/plugin-nft-generation/src/actions/mintNFTAction.ts:
```diff
@@ -0,0 +1,338 @@
+                template: mintTemplate,
+            });
+
+            const content = await generateObjectDeprecated({
```
We should be using generateObject and templated type safe schema's se plugin-coinbase for context

xwxtwd replied:
Thanksfor your guidance.

monilpat commented on packages/plugin-nft-generation/src/actions/mintNFTAction.ts:
```diff
@@ -0,0 +1,338 @@
+            console.log(e);
+        }
+
+        // callback();
```
can remove

xwxtwd replied:
fixed

mcp97 reviewed (changes requested):
Thanks for working on this a few comments otherwise looking good :)

odilitime committed:
Merge pull request #1525 from Titan-Node/add-livepeer-image-gen-1

feat: Add Livepeer Image Provider

shakkernerd committed:
Merge branch 'develop' into main

odilitime committed:
remove unused var

shakkernerd committed:
Merge pull request #839 from tsubasakong/main

feat: Add Custom System Prompt Support for plugin-image-generation

shakkernerd committed:
chore: add eslint support

shakkernerd committed:
chore: let -> const

odilitime committed:
remove unused vars per lint

shakkernerd committed:
chore: remove unused imports

odilitime committed:
Merge branch 'develop' into odi-fix

shakkernerd committed:
fix: unused variable

shakkernerd committed:
fix: unused variable

odilitime committed:
Merge pull request #1529 from elizaOS/odi-fix

chore: remove unused vars

shakkernerd committed:
fix: unused variable

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
Merge branch 'develop' into pr-842

shakkernerd committed:
Merge branch 'develop' into pr-842

shakkernerd committed:
chore: add eslint support

shakkernerd committed:
chore: let -> const

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge branch 'develop' into pr-1126

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
chore: bump version to v.0.1.7-alpha.2

shakkernerd committed:
Merge pull request #1530 from elizaOS/new_version_0.1.7-alpha.2

chore: bump version to v.0.1.7-alpha.2

shakkernerd committed:
Merge pull request #1519 from elizaOS/develop

chore: 0.1.7-alpha.2 prep, develop => main

actions-user committed:
chore: update changelog

e-fu committed:
fix(client-twitter): tighten engagement criteria and raise quality thresholds

e-fu committed:
Merge branch 'main' into fix/twitter-engagement-criteria

SK1989sL committed:
Update README.md

RedHorse823 committed:
Update README_CN - Model Configuration

The original model configuration guide is misleading, it tells readers to configure model provider by change the `XAI_MODEL` parameters.

But this is for the XAI configurations, i update the description completely with the right configuration steps.

swizzmagik committed:
Merge remote-tracking branch 'upstream/Main'

yanushevitz committed:
Added Polish Readme file

yanushevitz committed:
Fixed polish's readme typo's

yanushevitz committed:
Rename Readme_PL.md to README_PL.md

swizzmagik committed:
feat: fix chat client to autoscroll

swizzmagik committed:
chore: update changelog

mcp97 committed:
Merge pull request #1538 from swizzmagik/feat/add-autoscroll-chat-client

mcp97 committed:
Merge branch 'develop' into Polish_readme

mcp97 committed:
Merge pull request #1537 from yanushevitz/Polish_readme

Polish readme

mcp97 committed:
Merge branch 'develop' into Update-ReadMe-CN

mcp97 committed:
Merge pull request #1535 from RedHorse823/Update-ReadMe-CN

Update README_CN - Model Configuration

mcp97 committed:
Merge branch 'develop' into patch-1

odilitime committed:
Merge branch 'develop' into fix/twitter-engagement-criteria

mcp97 committed:
Merge pull request #1534 from SK1989sL/patch-1

docs: fix grammar/typos in README

mcp97 committed:
feat: improve messages

Improve message so criteria is displayed

mcp97 committed:
Merge branch 'develop' into fix/twitter-engagement-criteria

mcp97 committed:
Merge branch 'develop' into realitySpiral/dexExp

mcp97 committed:
Merge pull request #1533 from e-fu/fix/twitter-engagement-criteria

Fix: make twitter engagement criteria in prompt more focused on relevance to topics/interests/character

mcp97 committed:
Merge branch 'develop' into realitySpiral/dexExp

odilitime committed:
fix number prefixed username, add X brand in messaging

odilitime committed:
Merge branch 'develop' into twitter-fix

odilitime committed:
Merge pull request #1540 from elizaOS/realitySpiral/dexExp

feat: improve stale pr/issue messages

odilitime committed:
Merge branch 'develop' into twitter-fix

odilitime committed:
Merge pull request #1541 from odilitime/twitter-fix

fix: twitter usernames can start with numbers

mgunnin committed:
refactor: reorganize imports in video service

mgunnin committed:
fix: improve clarity and grammar in pull request template

azep-ninja committed:
fix tg markdown issue.

Ed-Marcavage committed:
fix: update jsdoc automation workflow

Ed-Marcavage committed:
fix: update jsdoc automation workflow

Ed-Marcavage committed:
Merge branch 'develop' into fix/jsdoc-automation

madjin committed:
Merge pull request #1547 from Ed-Marcavage/fix/jsdoc-automation

Fix jsdoc automation build issue & parametrize PR branch target

shakkernerd committed:
Merge branch 'develop' into fix--Telegram-Markdown-issues

madjin committed:
docs: Add JSDoc documentation to constants.ts

madjin committed:
docs: Add JSDoc documentation to createRuntime.ts

madjin committed:
docs: Add JSDoc documentation to testSetup.ts

madjin committed:
docs: Add JSDoc documentation to types.ts

mcp97 committed:
Merge pull request #1546 from azep-ninja/fix--Telegram-Markdown-issues

fix: Telegram Bad Request: can't parse entities

mcp97 committed:
Merge branch 'develop' into code-cleanup

madjin committed:
docs: Add JSDoc documentation to constants.ts

madjin committed:
docs: Add JSDoc documentation to createRuntime.ts

madjin committed:
docs: Add JSDoc documentation to types.ts

madjin committed:
docs: Add JSDoc documentation to testSetup.ts

shakkernerd committed:
Merge branch 'develop' into docs-update-full-1735428826331

mcp97 committed:
Merge pull request #1545 from mgunnin/code-cleanup

Chore: Refactor Imports. Improve PR Template

madjin committed:
Merge branch 'develop' into docs-update-full-1735429229223

mcp97 committed:
Merge branch 'develop' into docs-update-full-1735428826331

mcp97 committed:
Merge pull request #1549 from elizaOS/docs-update-full-1735429229223

docs: Add JSDoc documentation

mcp97 committed:
Merge branch 'develop' into docs-update-full-1735428826331

mcp97 committed:
Merge pull request #1548 from elizaOS/docs-update-full-1735428826331

docs: Add JSDoc documentation

slkzgm committed:
feat: Twitter spaces integration

slkzgm committed:
Adding pnpm-lock.yaml

HashWarlock committed:
add character file script to make system prompt and templates easier to add to existing character files or generate a new one

xwxtwd committed:
Merge branch 'develop' into feat/mint-NFT-with-collection-address

zkfriendly committed:
feat: add theme toggle functionality with dark and light mode support

- Imported ThemeToggle component in AppSidebar
- Added ThemeToggle to SidebarFooter in AppSidebar
- Initialized theme based on localStorage and system preferences in main.tsx to support dark and light modes

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1555 from zkfriendly/main

feat: add theme toggle functionality with dark and light mode support

YoungPhlo committed:
docs: Add "What Did You Get Done This Week? 7" notes

mcp97 committed:
Merge pull request #1559 from YoungPhlo/docs/community-streams

docs: Add "What Did You Get Done This Week? #7" notes

slkzgm committed:
Adding back comments and documentation

shakkernerd committed:
Merge branch 'develop' into pr-1554

xwxtwd committed:
Merge branch 'develop' into feat/mint-NFT-with-collection-address

samwhosung committed:
add TEE plugin-env support

samwhosung committed:
fix

samwhosung committed:
fix

L-jasmine committed:
fix: init 768 dimension in database for gaianet

shakkernerd committed:
chore: parse files through prettier

shakkernerd committed:
Merge pull request #1573 from elizaOS/enhance/update_script

feat: parse files through prettier

shakkernerd committed:
Merge branch 'pr-1572' into develop

tomguluson92 committed:
Update README.md

tomguluson92 committed:
Add files via upload

odilitime committed:
Merge branch 'develop' into patch-4

odilitime committed:
Merge pull request #1574 from tomguluson92/patch-4

docs: overview and preview of technical report

odilitime committed:
Merge branch 'develop' into samuv/env-tee

odilitime committed:
Merge pull request #1571 from samuveth/samuv/env-tee

feat: add TEE support for plugin-env

slkzgm committed:
feat: add optional TRANSCRIPTION_PROVIDER setting with fallback logic, moving twitter spaces plugins to Eliza repo

cmadaan committed:
Updates for web search and timeout

odilitime committed:
Merge branch 'develop' into evelyn

slkzgm committed:
Cleanup on packages/client-twitter/src/plugins/SttTtsPlugin.ts

yorkerhodes3 committed:
Update agents.md

Minor update to correct "should"

treppers committed:
add an client-direct endpoint to get memories by agentid and roomid

mcp97 committed:
Merge branch 'develop' into patch-1

zhourunlai committed:
update volcengine model

twilwa committed:
Merge branch 'develop' into memories_endpoint

0xFloyd committed:
docs: fix Contributing Guide

chandiniv1 committed:
feat: add docs for image generation plugin

salazarsebas committed:
fix: fix the image redirection

salazarsebas committed:
feat: Add Spanish Translation for Documentation README

mcp97 committed:
Merge pull request #1579 from yorkerhodes3/patch-1

Update agents.md - minor update

mcp97 committed:
Merge branch 'develop' into cleanup/documentation

ileana-pr committed:
added README files to all plugins

zhourunlai committed:
use tavily sdk

ileana-pr committed:
file updates

mcp97 committed:
Merge branch 'develop' into feat/add-readme-spanish-version-in-docs

bendanzhentan committed:
refactor(plugin-conflux): output detailed invalid content

zkvm committed:
tweak of evm transfer template

zhourunlai committed:
add experimental telemetry model option

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
Merge remote-tracking branch 'origin/develop' into pr/bendanzhentan/1602

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #1602 from bendanzhentan/refactor-output-conflux-errors

refactor(plugin-conflux): output detailed invalid content

shakkernerd committed:
Merge branch 'develop' into fix-plugin-evm-transfer-action

shakkernerd committed:
Merge pull request #1604 from zkvm/fix-plugin-evm-transfer-action

fix: tweak transfer template of plugin-evm

shakkernerd committed:
Merge remote-tracking branch 'origin/develop' into pr/zhourunlai/1603

shakkernerd committed:
Merge pull request #1603 from zhourunlai/experimental_telemetry

feat: add experimental telemetry model option

shakkernerd committed:
Merge branch 'develop' into readme-plugin-files

shakkernerd committed:
Merge pull request #1601 from ileana-pr/readme-plugin-files

docs: README.md files for plugins

shakkernerd committed:
chore: fix conflicts

shakkernerd committed:
chore: add readme

shakkernerd committed:
fix: set publishedDate to optional as stated by the docs.

shakkernerd committed:
chore: move tavily package to the core

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
feat: add error handling for TAVILY_API_KEY key

shakkernerd committed:
Merge branch 'develop' into tavily_sdk

shakkernerd committed:
Merge pull request #1599 from zhourunlai/tavily_sdk

feat: use tavily sdk

shakkernerd committed:
Merge branch 'develop' into feat/add-readme-spanish-version-in-docs

shakkernerd committed:
Merge pull request #1594 from salazarsebas/feat/add-readme-spanish-version-in-docs

docs: add readme spanish version in docs

shakkernerd committed:
Merge branch 'develop' into chandini/image-generaration-plugin

shakkernerd committed:
Merge pull request #1591 from chandiniv1/chandini/image-generaration-plugin

feat: add docs for image generation plugin

shakkernerd committed:
chore: remove conflicting files

shakkernerd committed:
chore: README files

shakkernerd committed:
chore: Update local-development.md

shakkernerd committed:
Merge branch 'develop' into cleanup/documentation

shakkernerd committed:
Merge pull request #1589 from 0xFloyd/cleanup/documentation

docs: fix Contributing Guide

shakkernerd committed:
Merge branch 'develop' into volcengine_model

shakkernerd committed:
Merge pull request #1586 from zhourunlai/volcengine_model

feat: update volcengine model

azep-ninja committed:
fix continue double responses

azep-ninja committed:
fix google api key

shakkernerd committed:
feat: include priority to plugin-tee build

shakkernerd committed:
chore: remove build cache with clean command

shakkernerd committed:
chore: changes to settings

shakkernerd committed:
fix: add @tavily/core to external libs (almost took my life)

shakkernerd committed:
chore: remove logs

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
Merge pull request #1609 from elizaOS/fix/breath_taking_error

fix: breath taking error

shakkernerd committed:
Merge branch 'develop' into fix/google-api-key-character-file

shakkernerd committed:
Merge pull request #1607 from azep-ninja/fix/google-api-key-character-file

fix: Google API Key not passing from character file

azep-ninja committed:
Merge branch 'develop' into fix/continue-double-response

cole-gillespie committed:
add plugins to the key components section of the faq

during a recent discussion on discord it was not clear to some folks the conceptual difference between clients and plugins. this just adds a bit more high level info up top on the FAQ for future creators.

cole-gillespie committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge pull request #1614 from cole-gillespie/patch-1

add plugins to the key components section of the faq

mcp97 committed:
Merge branch 'develop' into memories_endpoint

mcp97 committed:
Merge branch 'develop' into fix/continue-double-response

mcp97 committed:
Merge pull request #1581 from treppers/memories_endpoint

add an client-direct endpoint to get memories by agentid and roomid

azep-ninja committed:
Merge branch 'develop' into fix/continue-double-response

azep-ninja committed:
re-build fx

mcp97 committed:
Merge pull request #1606 from azep-ninja/fix/continue-double-response

fix: Double Responses from Continue Action

Commit:
Fix double spaced tweets in Post.ts

Currently, despite the prompting stating to use "\\n\\n (double spaces" between statements the tweets always output new statements with just a single newline, making tweets look blocky. 

For example, tweets previous would look like (despite the prompt): 
"Dobby is so excited to start this new year, full of possibilities and chances to make a difference!
Dobby has big plans to help his dear friends and make the wizarding world an even brighter place."

This ensures the double spaces requested in prompting is applied to the tweets with the tweets now looking like: 
"Dobby must remind all friends to be cautious of dark magic lurking in the shadows.

Dobby's heart aches at the thought of anyone falling prey to its deceitful powers."

This allows agents to tweet better-formatted posts that ease visilibty.

tcm390 committed:
add TranscriptionProvider

tcm390 committed:
select transcription provider based on character setting

tcm390 committed:
clean code

mcp97 committed:
Merge branch 'develop' into newline/tweets-newline-update

mcp97 committed:
Merge pull request #1626 from suicidalgoofy/newline/tweets-newline-update

Fix double spaced tweets in Post.ts

odilitime committed:
Merge branch 'develop' into tcm-transcription-config

mcp97 committed:
Merge pull request #1625 from elizaOS/tcm-transcription-config

feat: Select a transcription provider based on the character settings.

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into test-eliza

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into test-eliza

odilitime committed:
fix evm deps

odilitime committed:
fix bad merge

odilitime committed:
remove claude changes

mcp97 committed:
Merge pull request #1627 from odilitime/test-eliza

fix: turbo deps for plugin-evm

slkzgm committed:
fix: Merge conflicts with develop

lalalune committed:
Merge pull request #1550 from slkzgm/main

feat: Twitter Spaces Integration

ShreyGanatra committed:
corrected path for image upload

xMariem committed:
Add README_AR.md

Add Arabic language for readme.

Commit:
chore: update viem dependency version in plugin-evm and plugin-goat

mcp97 committed:
Merge branch 'develop' into ShreyGanatra/ImageUpload

xMariem committed:
Update README.md

add Arabic language reference

mcp97 committed:
Merge pull request #1632 from ShreyGanatra/ShreyGanatra/ImageUpload

fix: corrected path for image upload

mcp97 committed:
Update package.json

mcp97 committed:
Update package.json

mcp97 committed:
Merge branch 'develop' into update-viem-arthera

mcp97 committed:
Merge pull request #1637 from bertux/update-viem-arthera

chore: update viem dependency version in plugin-evm and plugin-goat

denizekiz committed:
Seperated imageModelProvider and imageVisionModelProvider for ImageDescriptionService

shakkernerd committed:
Merge branch 'develop' into evelyn

shakkernerd committed:
fix: remove unwanted gitignore

shakkernerd committed:
chore: add enable web search variable

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: import web search plugin only on when enabled

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
fix: lockfile

mcp97 committed:
testing

mcp97 committed:
update

mdominikd committed:
Added Hungarian README

mcp97 committed:
update clean.sh to include deleting lock file to really reset the repo

mcp97 committed:
update integreation test

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
run on pull request not pull request target

mcp97 committed:
update to use pnpm command

mcp97 committed:
trigger re-run of workflows

mcp97 committed:
fix

mcp97 committed:
revert

mcp97 committed:
actually need to revert

metakai1 committed:
Fix bug in plugin-bootstrap/src/evaluators/facts.ts

{{user1}} should be {{user2}} :

-------
in factEvaluator, in the examples template,

{{user1}} should be {{user2}} instead

                {
                    user: "{{user1}}",
                    content: { text: "Which city?" },
                },
                {
                    user: "{{user2}}",
                    content: { text: "Oakland" },
                },
                {
                    user: "{{user1}}",
                    content: {
                        text: "Oh, I've never been there, but I know it's in California",
                    },
                }

mcp97 committed:
standardize viem dependency so project builds now

mcp97 committed:
missing depdencies

mcp97 committed:
revert to target due to https://github.com/ai16z/eliza/pull/993/files#r1883351009

mcp97 committed:
Merge branch 'develop' into 711--fix-bug-in-facts_ts

mcp97 committed:
remove submodule

nusk0 committed:
Solve Issue 1643, imageDescriptionService not working with other providers than OpenAI

denizekiz committed:
modelVisionProvider Added to IagentRuntime interface type.ts

0xRider committed:
Update schema.sql

There is an extra semicolon that causes a syntax error when trying to upload the schema to supabase.

shakkernerd committed:
Merge pull request #1642 from Sifchain/realitySpiral/fixLockFile

fix: lockfile wasn't updated after dependency + bring viem to root level package.json

0xRider committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge branch 'develop' into 711--fix-bug-in-facts_ts

nusk0 committed:
Solve Issue 1643, imageDescriptionService not working with other providers than OpenAI

mcp97 committed:
Merge pull request #1648 from metakai1/711--fix-bug-in-facts_ts

fix: Fix bug in plugin-bootstrap/src/evaluators/facts.ts

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1634 from xMariem/main

Add README_AR.md

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1645 from mdominikd/main

Added Hungarian README

shakkernerd committed:
Merge branch 'develop' into pr/cmadaanaya/1577

shakkernerd committed:
Merge pull request #1577 from cmadaanaya/evelyn

fix: activate web-search plugin in agents/

mcp97 committed:
Merge branch 'develop' into 1634--fix_ImageDescriptionService

mcp97 committed:
Merge pull request #1667 from nusk0/1634--fix_ImageDescriptionService

fix: 1634  fix image description service

mcp97 committed:
Merge branch 'develop' into VisionModel2

mcp97 committed:
Merge pull request #1664 from denizekiz/VisionModel2

fix: Seperated imageModelProvider and imageVisionModelProvider for ImageDescriptioServices

mcp97 committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge pull request #1660 from 0xRider/patch-1

fix: Update Supabase schema.sql

odilitime committed:
Merge branch 'develop' into feat/mint-NFT-with-collection-address
